packages/sdk/wallets/wallet-walletconnect/src/wc-wallet.ts

```diff
@@ -0,0 +1,151 @@
+}
+
+/*
+ *   WARNING: This class should only be used with well-permissioned providers (ie IPC)
+ *   to avoid sensitive user 'privateKey' and 'passphrase' information being exposed
+ */
+export class WalletConnectWallet extends RemoteWallet<WalletConnectSigner> {
+  private options: ClientOptions
+  private metadata: SessionTypes.Metadata
+
+  private client?: WalletConnect
+  private pairing?: PairingTypes.Proposal
+  private session?: SessionTypes.Settled
+
+  constructor({ options, metadata }: { options?: ClientOptions; metadata: SessionTypes.Metadata }) {
+    super()
+
+    this.options = { relayProvider: 'wss://staging.walletconnect.org', ...options }
+    this.metadata = metadata
+
+    this.setupClient()
+  }
+
+  /**
+   * Get the URI needed for out of band session establishment
+   */
+  public async getUri() {
+    await waitForTruthy(() => this.pairing)
+    return this.pairing!.signal.params.uri
+  }
+
+  onSessionProposal = (proposal: SessionTypes.Proposal) => {
+    console.log('onSessionProposal', proposal)
+  }
+  onSessionCreated = (session: SessionTypes.Created) => {
+    console.log('onSessionCreated', session)
+  }
+  onSessionUpdated = (session: SessionTypes.Update) => {
+    console.log('onSessionUpdated', session)
+  }
+  onSessionDeleted = (session: SessionTypes.DeleteParams) => {
+    console.log('onSessionDeleted', session)
+  }
+
+  onPairingProposal = (pairing: PairingTypes.Proposal) => {
+    this.pairing = pairing
+  }
+  onPairingCreated = (pairing: PairingTypes.Created) => {
+    console.log('onPairingCreated', pairing)
+  }
+  onPairingUpdated = (pairing: PairingTypes.Update) => {
+    console.log('onPairingUpdated', pairing)
+  }
+  onPairingDeleted = (pairing: PairingTypes.DeleteParams) => {
+    console.log('onPairingDeleted', pairing)
+  }
+
+  private async setupClient() {
+    this.client = await WalletConnect.init(this.options)
+
+    this.client.on(CLIENT_EVENTS.session.proposal, this.onSessionProposal)
+    this.client.on(CLIENT_EVENTS.session.created, this.onSessionCreated)
+    this.client.on(CLIENT_EVENTS.session.updated, this.onSessionUpdated)
+    this.client.on(CLIENT_EVENTS.session.deleted, this.onSessionDeleted)
+
+    this.client.on(CLIENT_EVENTS.pairing.proposal, this.onPairingProposal)
+    this.client.on(CLIENT_EVENTS.pairing.created, this.onPairingCreated)
+    this.client.on(CLIENT_EVENTS.pairing.updated, this.onPairingUpdated)
+    this.client.on(CLIENT_EVENTS.pairing.deleted, this.onPairingDeleted)
+
+    // @ts-ignore no-dangling-promise
+    this.client
+      .connect({
+        metadata: this.metadata,
+        permissions: {
+          blockchain: {
+            // alfajores, mainnet, baklava
+            chains: ['celo:44787', 'celo:42220', 'celo:62320'],
+          },
+          jsonrpc: {
+            methods: Object.values(SupportedMethods),
+          },
+        },
+      })
+      .then((session) => (this.session = session))
```

Comment:
Recommend using `await`, promise `.then`s swallow errors

Reply:
Yeah this is due to some hoops I'm jumping through with establishing connections. Mentioned on Slack but will have another pass at it to try get a cleaner API